homeassistant/components/eight_sleep/sensor.py

```diff
@@ -284,6 +260,10 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
 class EightRoomSensor(EightSleepBaseEntity, SensorEntity):
     """Representation of an eight sleep room sensor."""
 
+    _attr_icon = "mdi:thermometer"
+    _attr_device_class = SensorDeviceClass.TEMPERATURE
+    _attr_native_unit_of_measurement = TEMP_CELSIUS
```

Comment:
Shouldn't `native_value` be updated here also for `EightRoomSensor`?

Reply:
And as a follow-up, I think you could even drop passing the "units" around

Reply:
`native_value` changes based on updates from the coordinator so I think it's fine to remain as a property. Good point about units, I thought I removed that logic but may not have been careful enough in my edits, will correct